### PR TITLE
Fix GitHub prepare action

### PR DESCRIPTION
**Pull request by hgy59**
## Description

This PR was initiated by the fact, that the prepare job of the github build action takes a large amount of time.
Local analysis showed that it takes > 60 seconds to evaluate the dependency-list for each package that depends on ffmpeg.

before, the "Evaluate dependencies" step of the "Prepare for Build" job took about 6 to 10 minutes.
with this fix, it takes about 3 minutes.

## Checklist

- [x] Build rule `all-supported` completed successfully
- [ ] New installation of package completed successfully
- [ ] Package upgrade completed successfully (Manually install the package again)
- [ ] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Includes small framework changes

### CODE REVIEW

File: cross/chromaprint/Makefile [hunk 1]
```diff
@@ -9,13 +9,13 @@ HOMEPAGE = https://acoustid.org/chromaprint
...
 
-OPTIONAL_DEPENDS = cross/ffmpeg4
+OPTIONAL_DEPENDS = cross/ffmpeg$(FFMPEG_VERSION)
 
 # compiler too old
 UNSUPPORTED_ARCHS = $(ARMv5_ARCHS) $(OLD_PPC_ARCHS)
 
 ifneq ($(wildcard $(FFMPEG_DIR)),)
-CMAKE_RPATH = /var/packages/ffmpeg$(FFMPEG_VERSION)/target/lib
+CMAKE_RPATH = /var/packages/ffmpeg$(subst 4,,$(FFMPEG_VERSION))/target/lib
```
Comment (hgy59) [inline cross/chromaprint/Makefile:18]: @th0ma7 I suppose without this fix the chromaprint package is not able to find the ffmpeg libs at runtime.

Comment (th0ma7) [reply cross/chromaprint/Makefile:18]: You are totally right, good catch.

File: .github/actions/prepare.sh [hunk 2]
```diff
@@ -37,17 +37,20 @@ done
...
 if [ "$(echo ${SPK_TO_BUILD} | grep -ow nzbdrone)" != "" ]; then
-    SPK_TO_BUILD+=" sonarr3"
+    SPK_TO_BUILD=$(echo "${SPK_TO_BUILD}" | tr ' ' '\n' | grep -v "nzbdrone" | tr '\n' ' ')" sonarr3"
 fi
 if [ "$(echo ${SPK_TO_BUILD} | grep -ow python)" != "" ]; then
-    SPK_TO_BUILD+=" python2"
+    SPK_TO_BUILD=$(echo "${SPK_TO_BUILD}" | tr ' ' '\n' | grep -v "python" | tr '\n' ' ')" python2"
+fi
+if [ "$(echo ${SPK_TO_BUILD} | grep -ow ffmpeg)" != "" ]; then
+    SPK_TO_BUILD=$(echo "${SPK_TO_BUILD}" | tr ' ' '\n' | grep -v "ffmpeg" | tr '\n' ' ')" ffmpeg4"
```
Comment (th0ma7) [inline .github/actions/prepare.sh:53]: If I get this right, you're removing all `ffmpeg{456}` from the list but only adding `ffmpeg4` at the end of list?  That would then by default disable `ffmpeg{56}` builds, no?

Comment (hgy59) [reply .github/actions/prepare.sh:53]: No, this is not the case, it just renames ffmpeg to ffmpeg4. (`-w ffmpeg` is a full word filter and does not match any of ffmpeg4, ffmpeg5 and ffmpeg6).

The error that is fixed with this is the following:
The modification of a file in the spk/ffmpeg4 folder triggers the build of ffmpeg package (not ffmpeg4) and I saw errors saying the folder spk/ffmpeg does not exist.
This custom handling is required for all packages where the package name does not match the spk folder (see lines above for python2 and sonarr3).

Comment (th0ma7) [reply .github/actions/prepare.sh:53]: Makes full sense, another nice catch :)

**Pull request merged**